src/browserlib/extract-events.mjs

```diff
@@ -0,0 +1,278 @@
+	  const iface = table.querySelector(`tr:nth-child(${interfaceRow + 1}) td:nth-child(2)`)?.textContent?.trim();
+	  if (eventName) {
+	    events.push({type: eventName, interface: iface, bubbles, src: { format: "css definition table", href: href(table.closest('*[id]')) }, href: href(table.closest('*[id]')) });
+	  }
+	}
+      }
+    });
+  }
+  // Look for the DOM-suggested sentence "Fire an event named X"
+  // or the Service Worker extension of "fire (a) functional event named"
+  [...document.querySelectorAll("a")].filter(a => !a.closest(informativeSelector)
+					     && (a.href === "https://dom.spec.whatwg.org/#concept-event-fire"
+						 || a.href === "https://w3c.github.io/ServiceWorker/#fire-functional-event" || a.href === "https://www.w3.org/TR/service-workers-1/#fire-functional-event-algorithm" || a.href === "https://www.w3.org/TR/service-workers-1/#fire-functional-event"
+						|| a.href === "https://w3c.github.io/pointerevents/#dfn-fire-a-pointer-event")
+					    )
+    .forEach(a => {
+      const container = a.parentNode;
+      let phrasing;
+      let m = container.textContent.match(/fir(e|ing)\s+a(n|\s+pointer)\s+event\s+named\s+"?(?<eventName>[a-z]+)/i);
+      if (m) {
+	if (m[2] === "n") {
+	  phrasing = "fire an event";
+	} else {
+	  phrasing = "fire a pointer event";
+	}
+      } else {
+	m = container.textContent.match(/fir(e|ing)\sa?\s*functional\s+event\s+(named\s+)?"?(?<eventName>[a-z]+)/i);
+	if (m) {
+	  phrasing = "fire functional event";
+	}
+      }
+      if (phrasing) {
+	const name = m.groups.eventName;
+	let newEvent = true;
+	let event = {src: { format: "fire an event phrasing", href: href(a.closest('*[id]')) }, href: href(a.closest('*[id]')) };
+	// this matches "fire an event named eventName" in battery-status and media capture main, named type in fullscreen, named e, event in html
+	if (name === 'eventName' || name === 'type' || name === 'e' || name === 'event') {
+	  return;
+	} else {
+	  event.type = name;
+	  // looking preferably for a or dfn elements, falling back to code
+	  const eventEl = [...container.querySelectorAll("a,dfn")].find(n => n.textContent.trim() === event.type) || [...container.querySelectorAll("code")].find(n => n.textContent.trim() === event.type);
+	  if (eventEl) {
+	    if (eventEl.tagName === "A" && eventEl.getAttribute("href")) {
+	      // If the event being fired is from another spec, let's skip it
+	      if (eventEl.getAttribute("href").startsWith("https://")) return;
+
+	      // otherwise, use the target of the link as our href
+	      event.href = eventEl.href;
+	    } else if (eventEl.tagName === "DFN" && eventEl.id) {
+	      eventEl.href = href(eventEl);
+	    }
+	    event.targets = fromEventElementToTargetInterfaces(eventEl);
+	  }
+	  // if we have already detected this combination, skip it
+	  if (events.find(e => isSameEvent(event, e))) {
+	    newEvent = false;
+	    event = events.find(e => isSameEvent(event, e));
+	  }
+	}
+	if (!event.interface) {
```

Comment:
To catch spec problems, shouldn't the code use the opportunity to assert that the info it may already have extracted from the definition table matches the interface info it gets from the prose and throw an error when that is not the case so that we know that the spec needs fixing?

Reply:
my expectation is that spec problems will be better caught by manual review of published events data (which in practice won't be changing all that often) than by making the already somewhat complex heuristics code try to be smarter.